core/src/androidMain/kotlin/io/github/aag/core/data/background/UpdateAuthTokenWorker.kt

```diff
@@ -0,0 +1,42 @@
+package io.github.aag.core.data.background
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import io.github.aag.core.domain.models.UserProfile
+import io.github.aag.core.domain.repositories.AuthRepository
+import io.github.aag.core.domain.repositories.ProfileRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class UpdateAuthTokenWorker(
+    appContext: Context,
+    workerParams: WorkerParameters,
+    private val authRepository: AuthRepository,
+    private val useProfileRepository: ProfileRepository
+) : CoroutineWorker(appContext = appContext, params = workerParams) {
+    private var userProfile: UserProfile? = null
+
+    init {
+        CoroutineScope(Dispatchers.IO + Job())
+            .launch {
+                useProfileRepository.getUserProfile()
+                    .collect { profile ->
+                        userProfile = profile
+                        doWork()
```

Comment:
I need to somehow trigger `doWork` on each `refreshToken` update

Reply:
Have you think about `StateFlow` with current `refreshToken` value inside repository?  `refreshTokenFlow.distinctUntilChanged().collect { authRepository.renewAuthToken(it) }` will be better

Reply:
`refreshToken` isn't supposed to be changed frequently, maybe not at all. While `authToken` gets expired on back-end every 2 hours